ZividOpenCV/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -0,0 +1,139 @@
+        const auto frame = camera.capture();
+
+        std::cout << "Setting up visualization" << std::endl;
+        Zivid::CloudVisualizer vis;
+        zivid.setDefaultComputeDevice(vis.computeDevice());
+
+        std::cout << "Displaying the point cloud" << std::endl;
+        vis.showMaximized();
+        vis.show(frame);
+        vis.resetToFit();
+
+        std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+        vis.run();
+
+        std::cout << "Converting ZDF point cloud to OpenCV format" << std::endl;
+
+        const auto pointCloud = frame.getPointCloud();
+        const auto rgb = pointCloudToRGB(pointCloud);
+
+        std::cout << "Undistorting the RGB image" << std::endl;
+
+        const auto cameraIntrinsticsCV = getCameraIntrinsics(camera.intrinsics());
+        const cv::Mat distortionCoefficients = std::get<0>(cameraIntrinsticsCV);
+        const cv::Mat cameraMatrix = std::get<1>(cameraIntrinsticsCV);
+
+        const cv::Size size = rgb.size();
+        const cv::Mat optimalCameraMatrix =
+            cv::getOptimalNewCameraMatrix(cameraMatrix, distortionCoefficients, size, 1, size);
+
+        cv::Mat rgbUndistorted;
+        cv::Mat rgbUndistortedFull;
+
+        cv::undistort(rgb, rgbUndistorted, cameraMatrix, distortionCoefficients);
+        cv::undistort(rgb, rgbUndistortedFull, cameraMatrix, distortionCoefficients, optimalCameraMatrix);
+
+        std::cout << "Displaying and saving the RGB image" << std::endl;
+
+        displayRGB(rgb, "Distorted RGB image");
+        cv::imwrite("Distorted RGB image.jpg", rgb);
+
+        std::cout << "Displaying and saving the undistorted RGB image" << std::endl;
+
+        displayRGB(rgbUndistorted, "Undistorted RGB image");
+        cv::imwrite("Undistorted RGB image.jpg", rgbUndistorted);
+
+        std::cout << "Displaying and saving the Undistorted RGB image - full" << std::endl;
+
+        displayRGB(rgbUndistortedFull, "Undistorted RGB image - full");
+        cv::imwrite("Undistorted RGB image - full.jpg", rgbUndistorted);
+    }
+    catch(const std::exception &e)
+    {
+        std::cerr << "Error: " << Zivid::toString(e) << std::endl;
+        return EXIT_FAILURE;
+    }
+}
+
+cv::Mat pointCloudToRGB(const Zivid::PointCloud &pointCloud)
+{
+    cv::Mat rgb(static_cast<int>(pointCloud.height()),
+                static_cast<int>(pointCloud.width()),
+                CV_8UC3,
+                cv::Scalar(0, 0, 0));
+
+    auto height = pointCloud.height();
+    auto width = pointCloud.width();
+
+    for(int i = 0; i < height; i++)
+    {
+        for(int j = 0; j < width; j++)
+        {
+            cv::Vec3b &color = rgb.at<cv::Vec3b>(i, j);
+            color[0] = pointCloud(i, j).blue();
+            color[1] = pointCloud(i, j).green();
+            color[2] = pointCloud(i, j).red();
+        }
+    }
+
+    return rgb;
+}
+
+std::tuple<cv::Mat, cv::Mat> getCameraIntrinsics(const Zivid::CameraIntrinsics &cameraIntrinsics)
```

Comment:
Can we think of a better name? We've kind of already gotten the `CameraIntrinsic`s, and are passing them to the function. So this function is more about converting it to to OpenCV formats.

Reply:
reformatCameraIntrinsics? I couldn't think of anything better. cameraIntrinsicsToOpenCV didn't look nice.